app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -337,8 +338,10 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         }
 
         navUrlInput.nextFocusDownId = when {
-            pocketVideosContainer.isVisible -> R.id.pocketVideoMegaTileView
-            else -> R.id.megaTileTryAgainButton
+            pocketVideosContainer.isEffectivelyVisible -> R.id.pocketVideoMegaTileView
+            megaTileTryAgainButton.isEffectivelyVisible -> R.id.megaTileTryAgainButton
+            tileContainer.adapter.itemCount == 0 -> R.id.navUrlInput
+            else -> R.id.tileContainer
```

Comment:
What does focusing the tileContainer do? Does it pass correctly to its descendants?

Reply:
I was as surprised as you are, but it does.  This is borrowed from `pocketVideoMegaTileView.nextFocusDownId` a few lines down.